pandas/_libs/tslibs/offsets.pyx

```diff
@@ -827,7 +824,34 @@ cpdef int get_day_of_month(datetime other, day_opt) except? -1:
         raise ValueError(day_opt)
 
 
-cpdef int roll_yearday(other, n, month, day_opt='start') except? -1:
+cpdef int roll_monthday(other, int n, compare):
+    """
+    Possibly increment or decrement the number of periods to shift
+    based on rollforward/rollbackward conventions.
+
+    Parameters
+    ----------
+    other : datetime or int
+    n : number of periods to increment, before adjusting for rolling
+    compare : datetime or int (must match `other`)
```

Comment:
I would really prefer not to do this, have separate functions. (compare)

Reply:
OK, am separating them now.  With some luck (and possibly help to confirm that the complexity in the `getOffsetOfMonth` methods is unneeded) we'll be able to get rid of the need to handle both cases in liboffsets.